application-tour-test/application-tour-test-pageobjects/src/main/java/org/xwiki/contrib/tour/test/po/TourFromLivetable.java

```diff
@@ -53,12 +56,17 @@ public boolean isActive()
         return isActive;
     }
     
+    public String getTargetClass()
+    {
+        return targetClass;
+    }
+    
     @Override
     public boolean equals(Object o)
     {
         if (o instanceof TourFromLivetable) {
             TourFromLivetable t = (TourFromLivetable) o;
-            return t.name.equals(name) && t.targetPage.equals(targetPage) && t.isActive == isActive;
+            return t.name.equals(name) && t.targetPage.equals(targetPage) && t.isActive == isActive && t.targetClass.equals(targetClass);
```

Comment:
What happens if targetClass is `null`?
You need to use EqualsBuilder:
Example: https://github.com/xwiki/xwiki-commons/blob/70e800371c7ff1d572b83800e44a0b8dfb26c659/xwiki-commons-core/xwiki-commons-extension/xwiki-commons-extension-api/src/main/java/org/xwiki/extension/AbstractExtensionDependency.java#L240-L240